File: backoffice/src/shared/domain/valueObjects/UuidValueObject.py
"""
 *
 * Libraries 
 *
"""

from uuid import uuid4

"""
 *
 * Class 
 *
"""

class UuidValueObject:

    """
     *
     * Attributes 
     *
    """

    __value : str

    """
     *
     * Methods 
     *
    """

    def __init__( self, value : str = None ) -> object:
        if value is None:
            value = self.__generateValue()
        self.__value = value
    
    def __generateValue( self ) -> str:
        return str( uuid4() )

    def getValue( self ) -> str:
        return self.__value

    def isEmpty( self ) -> bool:
        return self.__value == ''
    
    def equals( self, other : object ) -> bool:
        return self.__value == other.getValue()

File: preparation/src/shared/domain/valueObjects/UuidValueObject.cs
/* 
 * 
 * Class
 *
*/

namespace preparation.src.shared.domain {

    public class UuidValueObject {

        /* 
         * 
         * Attributes
         *
        */

        private string value;

        /* 
         * 
         * Methods
         *
        */

        public UuidValueObject( string value ) {
            this.value = value;
        }

        public UuidValueObject() {
            value = generateValue();
        }

        private string generateValue() {
            return Guid.NewGuid().ToString();
        }

        public string getValue() {
            return value;
        }

        public bool isEmpty() {
            return value == string.Empty;
        }

        public bool equals( UuidValueObject other ) {
            return value == other.getValue();
        }

    }

}